diff --git a/src/HW_models/NHW_NVMC.c b/src/HW_models/NHW_NVMC.c
index ec2c027eba9e..fff661c73c27 100644
--- a/src/HW_models/NHW_NVMC.c
+++ b/src/HW_models/NHW_NVMC.c
@@ -627,7 +627,7 @@ void nhw_nmvc_read_buffer(void *dest, uint32_t address, size_t size) {
     }
   }

-  if (offset + size >= backend->size) {
+  if (offset + size > backend->size) {
     OUT_OF_FLASH_ERROR(address + size);
   }

diff --git a/zephyr/module.yml b/zephyr/module.yml
index eb317c3ce8a6..4ed35dd2011a 100644
--- a/zephyr/module.yml
+++ b/zephyr/module.yml
@@ -1,2 +1,4 @@
 build:
   cmake: .
+  depends:
+    - hal_nordic
